design-proposals/api-review.md

```diff
@@ -0,0 +1,162 @@
+
+
+## Motivation
+Since KubeVirt has reached v1 and has some APIs with v1 version, implementing an API review process
+ensures code quality, usability, stability, and fosters community collaboration, ultimately leading to a successful and
+sustainable software ecosystem.  
+
+More importantly, it allows for a common framework through which all stake-holders can discuss and decide API facing 
+changes in KubeVirt.
+
+## Goals
+
+- Establish a process for API reviews such that:
+  - contributors have a clear idea about how to implement an API facing change
+  - reviews have a guideline about the necessary checks required for a successful API facing change
+  - community has a guideline about how to handle API breakages upon upgrade
+  - propose necessary tools to make process easier and reliable, and avoid human errors
+
+## Non Goals
+- Upgrades breaking due to behavioural changes, e.g. change in implementation of a controller
+
+## Definition of Users
+- API Reviewers
+- Contributors
+
+## User Stories
+- As a user of KubeVirt project, I want KubeVirt to maintain an intuitive, stable and simple API, that can 
+   be used as a foundational block to build products and projects
+- As a contributor, I need guidance on the right way to approach API facing change. Ideally this guidance should include
+    all the steps design docs, contributing the change and post contribution steps
+- As a reviewer, I need to have a comprehensive list of checks needed for approving an API facing change
+- As a reviewer, ideally, I need to be able to leverage automation wherever possible to make reviewing easier
+
+## Repos
+- https://github.com/kubevirt/kubevirt
+
+# Design
+
+In order to achieve the stability, quality and consistency of APIs like the core Kubernetes APIs this document proposes
+the following changes:
+
+- For API reviewers: There should be a one or more engineers that review api breaking changes on a regular basis.
+- For contributors: There should be a guide explaining how to merge and api change
+- Tools and tests: Tools and automation that can be helpful to reduce human burden and errors to carry-out api changes
+
+More details on each of the aforementioned items is highlighted below in separate sections.
+
+## Contributors responsibilities
+- Contributors must explain why the API change is needed, which functionality it adds and who requested it. They should
+  include links to github issues, email threads or any other public document where users request this functionality.
+- Contributors should contact the users who requested the functionality via the channel they have used, to give them the
+  opportunity to review the API. We do not want to define an API that the user finds inconvenient.
+
+## Process for api reviewers
+Recent changes to reviewer guidelines recommend forming small groups in specific areas of expertise. sig-api is 
+one such group. This group will be responsible for:
+- Reviewing all the PRs with `kind/api-change` labels
```

Comment:
In order to make sure that what effect this will have, the implementation is staggered into phases. Phase 1 is where most of the pains of this new process will be evident. In order to solve for the pain, automation tools can be introduced in phase 2 and phase 3. With automation and the best practices guide, I think we can cater to the needed velocity of development.